stories/core/Tree.stories.tsx

```diff
@@ -0,0 +1,199 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { Story, Meta } from '@storybook/react';
+import React from 'react';
+import { Tree, TreeProps, TreeNode, Checkbox } from '../../src/core';
+import { action } from '@storybook/addon-actions';
+import { SvgPlaceholder } from '@itwin/itwinui-icons-react';
+import { TreeData } from '../../src/core/Tree/Tree';
+
+export default {
+  component: Tree,
+  argTypes: {
+    className: { control: { disable: true } },
+    style: { control: { disable: true } },
+  },
+  title: 'Core/Tree',
+} as Meta<TreeProps>;
+
+export const Basic: Story<TreeProps> = (args) => {
+  const onSelectedNodeChange = (nodeId: string, selected: boolean) => {
+    if (selected) {
+      action(`Selected node ${nodeId}`)();
+    } else {
+      action(`Unselected node ${nodeId}`)();
+    }
+  };
+
+  const onNodeExpanded = (nodeId: string, expanded: boolean) => {
+    if (expanded) {
+      action(`Expanded node ${nodeId}`)();
+    } else {
+      action(`Closed node ${nodeId}`)();
+    }
+  };
+
+  const [disabledNodes] = React.useState([
+    'Node 4',
+    'Node 3.0',
+    'Node 6',
+    'Node 10',
+  ]);
+
+  const onCheckboxSelected = (nodeId: string, checked: boolean) => {
+    if (checked) {
+      action(`Checked node: ${nodeId}`)();
+    } else {
+      action(`Unchecked node: ${nodeId}`)();
+    }
+  };
+
+  const generateItem = React.useCallback(
+    (index: number, parentRow = '', depth = 0): TreeData => {
+      const keyValue = parentRow ? `${parentRow}.${index}` : `${index}`;
+      return {
+        nodeId: `Node ${keyValue}`,
+        label: `Node ${keyValue}`,
+        subLabel: `Sublabel for Node ${keyValue}`,
+        isDisabled:
+          disabledNodes.findIndex((id) => id === `Node ${keyValue}`) != -1,
+        subnodes:
+          depth < 10
+            ? Array(Math.round(index % 5))
+                .fill(null)
+                .map((_, index) => generateItem(index, keyValue, depth + 1))
+            : [],
+      };
+    },
+    [disabledNodes],
+  );
+
+  const data = React.useMemo(
+    () =>
+      Array(50)
+        .fill(null)
+        .map((_, index) => generateItem(index)),
+    [generateItem],
+  );
+
+  return (
+    <Tree
+      data={data}
+      nodeRenderer={(props) => (
+        <TreeNode
```

Comment:
I feel like it would be nice to show an example with `React.memo` to prevent rerendering on each action. But maybe after everything is cleared up we can help you with that.